app/uk/gov/hmrc/helptosavefrontend/connectors/NSIConnector.scala

```diff
@@ -130,6 +131,18 @@ class NSIConnectorImpl @Inject() (conf: Configuration, metrics: Metrics) extends
       }
   }
 
+  override def test(userInfo: NSIUserInfo)(implicit hc: HeaderCarrier, ex: ExecutionContext): Result[Unit] = EitherT[Future, String, Unit]{
```

Comment:
i guess we need to add the implicit headers as well `(nsiUserInfoFormat, hc.copy(authorization = None), ec)` based on my recent changes or rebase your branch to latest master and see